cms/lib/xblock/test/test_tagging.py

```diff
@@ -0,0 +1,151 @@
+        self.aside_tag = 'difficulty_tag'
+        self.aside_tag_value = 'hard'
+
+        course = CourseFactory.create(default_store=ModuleStoreEnum.Type.split)
+        self.course = ItemFactory.create(
+            parent_location=course.location,
+            category="course",
+            display_name="Test course",
+        )
+        self.chapter = ItemFactory.create(
+            parent_location=self.course.location,
+            category='chapter',
+            display_name="Week 1",
+            publish_item=True,
+            start=datetime(2015, 3, 1, tzinfo=UTC),
+        )
+        self.sequential = ItemFactory.create(
+            parent_location=self.chapter.location,
+            category='sequential',
+            display_name="Lesson 1",
+            publish_item=True,
+            start=datetime(2015, 3, 1, tzinfo=UTC),
+        )
+        self.vertical = ItemFactory.create(
+            parent_location=self.sequential.location,
+            category='vertical',
+            display_name='Subsection 1',
+            publish_item=True,
+            start=datetime(2015, 4, 1, tzinfo=UTC),
+        )
+        self.problem = ItemFactory.create(
+            category="problem",
+            parent_location=self.vertical.location,
+            display_name="A Problem Block",
+            weight=1,
+            user_id=self.user.id,
+            publish_item=False,
+        )
+        self.video = ItemFactory.create(
+            parent_location=self.vertical.location,
+            category='video',
+            display_name='My Video',
+            user_id=self.user.id
+        )
+
+        config = StudioConfig.current()
+        config.enabled = True
+        config.save()
+
+    def test_aside_contains_tags(self):
+        """
+        Checks that available_tags list is not empty
+        """
+        self.assertGreater(len(StructuredTagsAside.available_tags), 0,
+                           "StructuredTagsAside should contains at least one available tag")
+
+    def test_preivew_html(self):
+        """
+        Checks that html for the StructuredTagsAside is generated correctly
+        """
+        request = RequestFactory().get('/dummy-url')
+        request.user = UserFactory()
+        request.session = {}
+
+        # Call get_preview_fragment directly.
+        context = {
+            'reorderable_items': set(),
+            'read_only': True
+        }
+        problem_html = get_preview_fragment(request, self.problem, context).content
+
+        # Check that aside is rendered correctly
+        self.assertRegexpMatches(problem_html, r'<select\s[_=a-zA-Z0-9\"\']+\sname=[\"\']%s[\"\']' % self.aside_tag)
+        self.assertRegexpMatches(problem_html, r"data-block-type=[\"\']%s[\"\']" % self.aside_name)
+        self.assertRegexpMatches(problem_html, r"data-init=[\"\']StructuredTagsInit[\"\']")
+        self.assertRegexpMatches(problem_html, r"data-runtime-class=[\"\']PreviewRuntime[\"\']")
```

Comment:
It would be better to test this by parsing the HTML with lxml, so that you can look for nodes with particular attributes, rather than using a regex (which is going to be vulnerable to ordering/spacing changes).